cpp/src/arrow/io/file.cc

```diff
@@ -412,7 +412,7 @@ class MemoryMappedFile::MemoryMap : public MutableBuffer {
 
   bool closed() const { return !file_->is_open(); }
 
-  Status Open(const std::string& path, FileMode::type mode) {
+  Status Open(const std::string& path, FileMode::type mode, const off_t offset) {
```

Comment:
I'd add a check that the offset is a multiple of PAGE_SIZE.

Reply:
Will add this logic. Thanks!